arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/DynamicClassExtractor.java

```diff
@@ -329,44 +320,36 @@ Object restoreResponse(Object result) {
     private boolean needRecord() {
         /*
          * Judge whether the hash value of the method signature has been recorded to avoid repeated recording.
-         * The nonparametric method may return different results and needs to be recorded
          * */
         ArexContext context = ContextManager.currentContext();
-        if (context != null && methodKey != null) {
+        if (context != null) {
             this.methodSignatureKey = buildDuplicateMethodKey();
-            this.methodSignatureKeyHash = StringUtil.encodeAndHash(methodSignatureKey);
-            if (context.getMethodSignatureHashList().contains(methodSignatureKeyHash)) {
+            if (methodKey != null) {
+                this.methodSignatureKeyHash = StringUtil.encodeAndHash(this.methodSignatureKey);
+            } else {
+                // no argument method check repeat by className + methodName + result
+                this.methodSignatureKeyHash = buildNoArgMethodSignatureHash();
+            }
+            if (context.getMethodSignatureHashList().contains(this.methodSignatureKeyHash)) {
                 if (Config.get().isEnableDebug()) {
                     LogManager.warn(NEED_RECORD_TITLE,
-                            StringUtil.format("do not record method, cuz exist same method signature: %s", methodSignatureKey));
+                            StringUtil.format("do not record method, cuz exist same method signature: %s", this.methodSignatureKey));
                 }
                 return false;
             }
         }
 
-        if (result == null || result instanceof Throwable) {
+        if (this.result == null || this.result instanceof Throwable) {
             return true;
         }
 
-        try {
-            int size = 0;
-            if (result instanceof Collection<?>) {
-                size = ((Collection<?>) result).size();
-            } else if (result instanceof Map<?, ?>) {
-                size = ((Map<?, ?>) result).size();
-            } else if (result.getClass().isArray()) {
-                size = Array.getLength(result);
-            }
-            if (size > RESULT_SIZE_MAX) {
-                String methodInfo = methodSignatureKey == null ? buildDuplicateMethodKey() : methodSignatureKey;
-                LogManager.warn(NEED_RECORD_TITLE,
-                        StringUtil.format("do not record method, cuz result size:%s > max limit: %s, method info: %s",
-                                String.valueOf(size), String.valueOf(RESULT_SIZE_MAX), methodInfo));
-                return false;
-            }
-        } catch (Throwable e) {
-            LogManager.warn(NEED_RECORD_TITLE, e);
+        if (!agentSizeOf.checkMemorySizeLimit(this.result, ArexConstants.MEMORY_SIZE_1MB)) {
```

Comment:
When the size exceeds the size, you can consider recording it as a null value to avoid fuzzy matching with other normal data during replay, which will affect problem troubleshooting. It is recommended to add a warning log.

Reply:
get!